samples/client-sdk/getting-started-daily-js/index.js

```diff
@@ -336,89 +412,81 @@ class DailyCallManager {
-    cameraPromptOption.disabled = true;
-    cameraSelector.appendChild(cameraPromptOption);
+    // Element references for camera and microphone selectors.
+    const selectors = {
+      videoinput: document.getElementById('camera-selector'),
+      audioinput: document.getElementById('mic-selector'),
+    };
 
-    const micPromptOption = new Option('Select a microphone', '', true, true);
-    micPromptOption.disabled = true;
-    micSelector.appendChild(micPromptOption);
+    // Prepare selectors by clearing existing options and adding a non-selectable prompt.
+    Object.values(selectors).forEach((selector) => {
+      selector.innerHTML = '';
+      const promptOption = new Option(
+        `Select a ${selector.id.includes('camera') ? 'camera' : 'microphone'}`,
+        '',
+        true,
+        true
+      );
+      promptOption.disabled = true;
+      selector.appendChild(promptOption);
+    });
 
-    // Populate the selectors and set currently selected devices as selected
+    // Create and append options to the selectors based on available devices.
     devices.forEach((device) => {
-      if (device.label) {
-        const option = new Option(device.label, device.deviceId);
-
-        if (device.kind === 'videoinput') {
-          cameraSelector.appendChild(option);
-          if (device.deviceId === currentCameraDeviceId) {
-            option.selected = true;
-          }
-        } else if (device.kind === 'audioinput') {
-          micSelector.appendChild(option);
-          if (device.deviceId === currentMicDeviceId) {
-            option.selected = true;
-          }
-        }
+      if (device.label && selectors[device.kind]) {
+        const isSelected =
+          selectedDevices[device.kind === 'videoinput' ? 'camera' : 'mic']
+            .deviceId === device.deviceId;
+        const option = new Option(
+          device.label,
+          device.deviceId,
+          isSelected,
+          isSelected
+        );
+        selectors[device.kind].appendChild(option);
       }
     });
 
-    // Attach event listeners to the selectors to handle device changes.
-    this.addDeviceChangeListener(cameraSelector, 'video');
-    this.addDeviceChangeListener(micSelector, 'audio');
-  }
-
-  /**
-   * Attaches a change event listener to a device selector.
-   * This listener updates the currently selected device
-   * for video or audio based on user selection.
-   *
-   * @param {HTMLElement} selector - The <select> element for choosing a device.
-   * @param {string} deviceType - The type of device, either 'video' for cameras or 'audio' for microphones.
-   */
-  addDeviceChangeListener(selector, deviceType) {
-    selector.addEventListener('change', (e) => {
-      const deviceId = e.target.value;
-      const deviceOptions = { [`${deviceType}DeviceId`]: deviceId };
-      this.call.setInputDevicesAsync(deviceOptions);
+    //Attach event listeners for device changes.
```

Comment:
nit: because when i first read this, i thought you were setting up a listener like we have already built in for when the underlying list of devices changes, rather than when the user selects something different in our generated UI


```suggestion
    // Listen for user device change requests.
```